employee-rostering-frontend/src/store/roster/operations.ts

```diff
@@ -39,9 +42,18 @@ interface KindaShiftRosterView extends Omit<ShiftRosterView, "spotIdToShiftViewL
   spotIdToShiftViewListMap: ObjectNumberMap<KindaShiftView[]>;
 }
 
+interface KindaAvailabilityRosterView extends Omit<AvailabilityRosterView, "employeeIdToShiftViewListMap" | "employeeIdToAvailabilityViewListMap" | "unassignedShiftViewList" > {
+  employeeIdToShiftViewListMap: ObjectNumberMap<KindaShiftView[]>;
+  employeeIdToAvailabilityViewListMap: ObjectNumberMap<KindaEmployeeAvailabilityView[]>;
+  unassignedShiftViewList: KindaShiftView[];
+}
+
 let lastCalledShiftRosterArgs: any | null;
 let lastCalledShiftRoster: ThunkCommandFactory<any, SetShiftRosterIsLoadingAction | SetShiftRosterViewAction> | null = null;
 
+let lastCalledAvailabilityRosterArgs: any | null;
+let lastCalledAvailabilityRoster: ThunkCommandFactory<any, SetAvailabilityRosterIsLoadingAction | SetAvailabilityRosterViewAction> | null = null;
```

Comment:
Is this the best way to achieve auto-refresh? Maybe if the TSX component set up the timeout it could use the same arguments and you wouldn't have to remember it here? Or you could keep it in the AppState. Any example code, blog etc. out there that uses similar approach?

Reply:
The primary issue of putting auto-refresh directly into a TSX component is that `Shift` and `EmployeeAvailability` operations modify both the Shift Roster and the Availability roster in non-trivial ways (i.e. can generate new indictments), meaning the Shift Roster and Availability roster need to be refreshed. There are quite a few ways to call `Shift` and `EmployeeAvailability` operations in the UI (using the `Modal` or click and drag a new Shift), so it cannot be tied to a single UI component.